File: uvm/api/com/untangle/uvm/logging/LoggingManager.java
/**
 * $Id$
 */
package com.untangle.uvm.logging;

/**
 * Manager for Logging.
 */
public interface LoggingManager
{
    /**
     * resetAllLogs re-reads all log configuration files (jboss,
     * uvm, and all app instances). This allows changing logging levels,
     * etc. The old output files will be erased and new files begun.
     */
    void resetAllLogs();

    /**
     * Set the logging context of this thread to the "uvm" configuration
     * log4j log calls after this will go to the uvm.log file
     */
    void setLoggingUvm();

    /**
     * Set the logging context of this thread to the "app" configuration
     * log4j log calls after this will go to the associated app-appId.log file
     */
    void setLoggingApp(Long appId);
}

File: uvm/hier/usr/lib/python2.7/uvm/logging_manager.py
from uvm.manager import Manager
import string

class LoggingManager(Manager):
    def __init__(self, uvmContext):
        self.__uvmContext = uvmContext
        self.__loggingManager = self.__uvmContext.loggingManager()

    def api_resetlogs(self):
        self.__loggingManager.resetAllLogs()

Manager.managers.append( LoggingManager )
